source/blog/2017-11-01-async-stack-traces.md

```diff
@@ -1,38 +1,42 @@
 ---
 layout: post
-title: Introducing Twitter Futures Capture Points
+title: 🛠️  Introducing Twitter Futures Capture Points
 published: true
 post_author:
   display_name: Stefan Lance
   twitter: stefan_lance
-tags: Util
+tags: Util, IntelliJIDEA
 ---
 
-If you've ever tried to debug asynchronous code written with Twitter `Future`s,
+If you've ever tried to debug asynchronous code written with Twitter [`Future`](https://github.com/twitter/util/blob/develop/util-core/src/main/scala/com/twitter/util/Future.scala)s, 
 you may have found that the stack trace isn't always helpful.  The stack trace
 at a given point in the program includes the active stack frames and thus shows
 the chain of function calls that led up to that point.  Since asynchronous
 execution causes different parts of a program to run on different threads, the
```

Comment:
i'd omit different threads. you can have this behavior on a single thread too. how about:

"Since asynchronous execution causes different parts of a program have their stacktraces chopped up, the stack trace often shows unexpected internal library calls and is missing the causality developers are used to from synchronous code."